manager/integration/tests/test_basic.py

```diff
@@ -3006,6 +3008,163 @@ def test_backup_lock_creation_during_deletion(set_random_backupstore, client, co
     _, b2 = common.find_backup(client, std_volume_name, snap2.name)
 
 
+def backup_ownership_transfer_when_instance_manager_is_down(client, core_api, volume_name, request, recover=False):  # NOQA
+    """
+    Scenario: test backup ownership transfer when the engine instance
+              manager is down.
+
+    Context: longhorn/longhorn#4064
+
+    Given a volume created, and attached.
+    And some data written to the volume.
+    And backup created.
+
+    When cordon the node with backup.
+    And delete the backup engine instance manager pod.
+        engine instance manager is in error state.
+
+    Then backup ownerID should change.
+    And wait for backup to complete.
+    And detach volume.
+    """
+    backupstore_cleanup(client)
+
+    volume_size = str(100 * Mi)
+    volume_replicas = 2
+    volume = create_and_check_volume(
+        client, volume_name, volume_replicas, size=volume_size
+    )
+
+    host_id = get_self_host_id()
+    volume = volume.attach(hostId=host_id)
+    volume = common.wait_for_volume_healthy(client, volume.name)
+
+    new_data = {
+        'pos': 0,
+        'content': generate_random_data(20 * Mi),
+    }
+
+    _, backup, snapshot, written_data = create_backup(
+        client, volume_name, data=new_data)
+
+    original_ownerID = backup.ownerID
+
+    instance_managers = client.list_instance_manager()
+    engine_instance_manager = None
+    for im in instance_managers:
+        if im.managerType != "engine":
+            continue
+
+        if im.nodeID != backup.ownerID:
+            continue
+
+        engine_instance_manager = im
+    assert engine_instance_manager is not None, \
+        f'expect to find engine instance manager for {backup.name}'
+
+    def finalizer():
+        set_node_cordon(core_api, engine_instance_manager.nodeID, False)
+
+    request.addfinalizer(finalizer)
+
+    set_node_cordon(core_api, engine_instance_manager.nodeID, True)
+
+    delete_and_wait_pod(
+        core_api,
+        engine_instance_manager.name,
+        namespace=common.LONGHORN_NAMESPACE
+    )
+
+    for _ in range(RETRY_COUNTS):
+        engine_instance_manager = client.by_id_instance_manager(
+            engine_instance_manager.name
+        )
+        if engine_instance_manager.currentState == "error":
+            break
+        time.sleep(RETRY_INTERVAL)
+    assert engine_instance_manager.currentState == "error"
+
+    for _ in range(RETRY_COUNTS):
+        _, backup = find_backup(client, volume.name, snapshot.name)
+        if backup.ownerID != original_ownerID:
+            break
+        time.sleep(RETRY_INTERVAL)
+
+    backup_volume, backup = find_backup(client, volume.name, snapshot.name)
+    assert backup.ownerID != original_ownerID, \
+        f'expecting {backup.name} ownerID {original_ownerID} to change'
+
+    wait_for_backup_completion(client, volume.name, snapshot.name)
+    volume.detach(hostId="")
+
+    if recover:
```

Comment:
the node will be uncordoned anyway because it will be done int he reqeust finalizer, why we need this here?